src/transaction/offset.rs

```diff
@@ -1,194 +0,0 @@
-use super::{TRANSACTION_CREATE_FIXED_SIZE, TRANSACTION_SCRIPT_FIXED_SIZE};
-use crate::{Metadata, StorageSlot, Transaction};
-
-use fuel_types::bytes::{self, SizedBytes};
-use fuel_types::Bytes32;
-
-impl Transaction {
```

Comment:
All offsets moved to `Create` and `Script` accordingly. But those offsets are relative to `Create` and `Script` instead of `Transaction`.
To get the relative offset to the `Transaction`, you need to add `Transaction::offset` to this offset.

For example, `Create::gas_price_offset` right now is `0`, when previously `Transaction::gas_price_offset`  was `8` because 8 bytes were reserved by default for `TransactionRepr`. So right now `Transaction::gas_price_offset` = `Create::gas_price_offset` + `Transaction::offset`.

Reply:
I decided to remove this change, and now all offset and serialization include 8 bytes for the `Transaction` enum discriminant.

So `Script` and `Create` are serialized as before the change. `Transaction` is only a wrapper that calls methods of inner type.

Maybe it is unclear why `Script` and `Create` have 8 bytes in the canonical serialization/deserialization, but it is simpler for the user and doesn't need to remember that offsets are relative. Also `serialization_size` and `metered_bytes` are correct=)